opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -227,7 +227,19 @@ namespace Opm {
 
         using namespace std::placeholders;
         const auto handler = std::bind( handleKW, _1, schedule, props, n_xyz );
+
+#ifdef _MSC_VER
+        for (auto& x : section)
+        {
+            auto keywords = handler(x);
+            for (auto& keyword : keywords)
+            {
+                this->keywords.push_back(keyword);
+            }
+        }
+#else
         this->keywords = fun::concat( fun::map( handler, section ) );
```

Comment:
I don't think that's an argument for using poor abstractions. I think we should rather improve upon and document better abstractions so they **are** know well. `map` and `concat` aren't conceptually hard, and what's going on and intent is communicated in an instant instead of having to mentally step through the program (loop in this case). The loop may have components that are easier to immediately recognise, which could be considered a case of Stockholm Syndrome I might add, but only in the sense that it's easier to see what they **do**, not what we want to **happen**. And I sincerely believe we should emphasise the latter.

If `map` and `concat` aren't sufficiently documented to be understood then that's a documentation problem and we'll fix that.

This problem should be addressed as a problem with the broken MSVC compiler and handled in `fun`. This `#ifdef` solution isn't happening regardless.